Mathlib/Topology/Algebra/FilterBasisNew.lean

```diff
@@ -0,0 +1,448 @@
+/-
+Copyright (c) 2021 Patrick Massot. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Patrick Massot, Anatole Dedecker
+-/
+import Mathlib.Order.Filter.Bases
+import Mathlib.Topology.Algebra.Module.Basic
+
+/-!
+**WARNING** : This is a new version of `Topology.Algebra.FilterBasis`, which is still used
+by the library at this point in time. The library will be gradually modified to use this file
+instead. See `https://leanprover.zulipchat.com/#narrow/channel/287929-mathlib4/topic/Refactoring.20algebraic.20filter.20bases/near/479437345`
+for more details.
+
+# Algebraic filter bases
+
+We know that the topology of a topological group (or ring, or module) is invariant by
+translations, hence it is completely described by its filter of neighborhoods at the
+neutral element (`1` for groups, `0` for additive groups, rings and modules).
+In this file, we completely characterize the filter bases which can be obtained as bases of
+neighborhoods of the neutral element for some group (or ring, or module) topology.
+
+More precisely, for `X` in `{Group, AddGroup, Ring, Module}`, we provide :
```

Comment:
What about monoid topologies?

Reply:
Well, monoid topologies are not described by their neighborhood filter at a single point... For example, for the additive monoid of nonnegative reals, the usual topology is a monoid topology, and so is (I believe) the topology generated by all `[a, b[` intervals. These two topologies are definitely distinct, but they have the same neighborhood filter at the origin.

Reply:
I guess it would still make sense to write down the basis version of [ContinuousAdd.of_nhds_zero](https://leanprover-community.github.io/mathlib4_docs/Mathlib/Topology/Algebra/Monoid.html#ContinuousAdd.of_nhds_zero), but it sounds a bit weird to me precisely because the hypotheses of that theorem are not true for any topological monoid (e.g not in `NNReal` with its usual topology)